unittests/ThunkLibs/generator.cpp

```diff
@@ -376,564 +372,506 @@ Fixture::GenOutput Fixture::run_thunkgen(std::string_view prelude, std::string_v
+    hasName("to_guest"),
+    // Parameter is a host_layout<A> (ignoring qualifiers and references)
+    hasParameter(0, hasType(references(classTemplateSpecializationDecl(hasName("host_layout"), hasAnyTemplateArgument(refersToType(asString("struct A"))))))),
+    // Return value is a guest_layout<A>
+    returns(asString("guest_layout<" CLANG_STRUCT_PREFIX "A>"))));
+  const auto guest_converter_undefined = matches(functionDecl(
+    hasName("to_guest"),
+    // Parameter is a host_layout<A> (ignoring qualifiers and references)
+    hasParameter(0, hasType(references(classTemplateSpecializationDecl(hasName("host_layout"), hasAnyTemplateArgument(refersToType(asString("struct A"))))))), isDeleted()));
+
+  const std::string code =
+    "template<typename> struct fex_gen_type {};\n"
+    "template<> struct fex_gen_type<A> {};\n";
+
+  // For fully compatible types, both guest_layout and host_layout directly
+  // reference the original struct
+  SECTION("Fully compatible type") {
+    const char* struct_def = "struct A { int a; int b; };\n";
+    const auto output = run_thunkgen_host(struct_def, code, guest_abi);
+    CHECK_THAT(output, matches(classTemplateSpecializationDecl(hasName("guest_layout"), hasAnyTemplateArgument(refersToType(asString("struct A"))), has(fieldDecl(hasName("data"), hasType(hasCanonicalType(asString("struct A"))))))));
+    CHECK_THAT(output, guest_converter_defined);
+
+    CHECK_THAT(output, host_layout_is_trivial);
+  }
+
+  // For repackable types, guest_layout explicitly lists its members
+  SECTION("Repackable type") {
+    const char* struct_def =
+      "#ifdef HOST\n"
+      "struct A { int a; int b; };\n"
+      "#else\n"
+      "struct A { int b; int a; };\n"
+      "#endif\n";
+    const auto output = run_thunkgen_host(struct_def, code, guest_abi);
+    CHECK_THAT(
+      output,
+      matches(classTemplateSpecializationDecl(
+        hasName("guest_layout"), hasAnyTemplateArgument(refersToType(asString("struct A"))),
+        // The member "data" exists and is defined to a struct...
+        has(fieldDecl(
+          hasName("data"),
+          hasType(hasCanonicalType(hasDeclaration(decl(
+            // ... the members of which also use guest_layout (with fixed-size integers)
+            has(fieldDecl(hasName("a"), hasType(asString("guest_layout<int32_t>")))), has(fieldDecl(hasName("b"), hasType(asString("guest_"
+                                                                                                                                   "layout<"
+                                                                                                                                   "int32_"
+                                                                                                                                   "t"
+                                                                                                                                   ">")))))))))))));
```

Comment:
Following up on https://github.com/FEX-Emu/FEX/pull/3413#issuecomment-1959081683:
> Why does it decide to break up the string several times instead of putting any of the has(...) expressions on a new line?

`PenaltyBreakString: 10` will fix this, but I haven't checked what ripple effects that has on the rest of FEX. Seems highly desirable in this particular instance though. (Note that clang-format won't un-break strings, so changes here need to be re-applied on the pre-format sources).

Reply:
I have removed the option and compared the trees with and without the option side by side and it seems that there's no situation where that option actually makes things better.

The only different I see that's more relevant are things like (in `eflags_signal.cpp`) for example:

![Screenshot from 2024-02-23 09-26-20](https://github.com/FEX-Emu/FEX/assets/7911/18abaa1f-4790-400c-b46b-b23f25612064)

I will regenerate the tree again.